chainer/optimizers/adam.py

```diff
@@ -62,12 +62,38 @@ def _inplace_axpby(x, a, b, y):
     if isinstance(x, intel64.mdarray):
         x.inplace_axpby(a, b, y)
     else:
-        if a == 1:
+        if isinstance(a, (int, float, numpy.generic)) and a == 1:
```

Comment:
In which case is this check needed?

Reply:
In a fused function, it is not allowed to write conditions which include scalar variables determined runtime. I'll write it as a comment.